internal/service/timestreaminfluxdb/db_instance.go

```diff
@@ -0,0 +1,1182 @@
+					to your workloads requirements: Influx IO Included 3000 IOPS, Influx IO Included 12000 IOPS, 
+					Influx IO Included 16000 IOPS. This argument is optional. Possible values: 
+					"InfluxIOIncludedT1", 
+					"InfluxIOIncludedT2", 
+					"InfluxIOIncludedT3". 
+					If not provided, defaults to "InfluxIOIncludedT1"`,
+			},
+			"deployment_type": schema.StringAttribute{
+				Optional: true,
+				Computed: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+				Validators: []validator.String{
+					stringvalidator.OneOf(
+						"SINGLE_AZ",
+						"WITH_MULTIAZ_STANDBY",
+					),
+				},
+				Description: `Specifies whether the DB instance will be deployed as a standalone instance or 
+					with a Multi-AZ standby for high availability. This argument is optional. 
+					Possible values: 
+					"SINGLE_AZ", 
+					"WITH_MULTIAZ_STANDBY". 
+					If not provided, defaults to "SINGLE_AZ"`,
+			},
+			"endpoint": schema.StringAttribute{
+				Computed: true,
+			},
+			"id":                                framework.IDAttribute(),
+			"influx_auth_parameters_secret_arn": framework.ARNAttributeComputedOnly(),
+			"name": schema.StringAttribute{
+				Required: true,
+				// TIP: ==== PLAN MODIFIERS ====
+				// Plan modifiers were introduced with Plugin-Framework to provide a mechanism
+				// for adjusting planned changes prior to apply. The planmodifier subpackage
+				// provides built-in modifiers for many common use cases such as
+				// requiring replacement on a value change ("ForceNew: true" in Plugin-SDK
+				// resources).
+				//
+				// See more:
+				// https://developer.hashicorp.com/terraform/plugin/framework/resources/plan-modification
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+				Validators: []validator.String{
+					stringvalidator.LengthAtLeast(3),
+					stringvalidator.LengthAtMost(40),
+					stringvalidator.RegexMatches(
+						// Taken from the model for TimestreamInfluxDB in AWS SDK Go V2
+						// https://github.com/aws/aws-sdk-go-v2/blob/8209abb7fa1aeb513228b4d8c1a459aeb6209d4d/codegen/sdk-codegen/aws-models/timestream-influxdb.json#L1215
+						regexache.MustCompile("^[a-zA-z][a-zA-Z0-9]*(-[a-zA-Z0-9]+)*$"),
+						"",
+					),
+				},
+			},
+			names.AttrTags:    tftags.TagsAttribute(),
+			names.AttrTagsAll: tftags.TagsAttributeComputedOnly(),
+			"organization": schema.StringAttribute{
+				Optional: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"password": schema.StringAttribute{
+				Required: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+				Validators: []validator.String{
+					stringvalidator.LengthAtLeast(8),
+					stringvalidator.LengthAtMost(64),
+					stringvalidator.RegexMatches(regexache.MustCompile("^[a-zA-Z0-9]+$"), ""),
```

Comment:
Surprised that only alphanumeric passwords are supported?

Reply:
I am somewhat surprised. The regex is defined in AWS SDK Go [here](https://github.com/aws/aws-sdk-go-v2/blob/8209abb7fa1aeb513228b4d8c1a459aeb6209d4d/codegen/sdk-codegen/aws-models/timestream-influxdb.json#L2293).